models/model_choice.py

```diff
@@ -99,81 +125,76 @@ def set_hyperparameters(params, num_classes, model, checkpoint, dontcare_val):
     step_size = get_key_def('step_size', params['training'], None, "missing mandatory step size parameter")
     gamma = get_key_def('gamma', params['training'], None, "missing mandatory gamma parameter")
 
-    # optional hyperparameters. Set to None if not in config file
-    class_weights = torch.tensor(params['training']['class_weights']) if params['training']['class_weights'] else None
-    if params['training']['class_weights']:
-        verify_weights(num_classes, class_weights)
+    class_weights = torch.tensor(class_weights) if class_weights else None
 
     # Loss function
-    criterion = MultiClassCriterion(loss_type=params['training']['loss_fn'],
+    criterion = MultiClassCriterion(loss_type=loss_fn,
                                     ignore_index=dontcare_val,
                                     weight=class_weights)
 
     # Optimizer
-    opt_fn = params['training']['optimizer']
+    opt_fn = optimizer
     optimizer = create_optimizer(params=model.parameters(), mode=opt_fn, base_lr=lr, weight_decay=weight_decay)
     lr_scheduler = optim.lr_scheduler.StepLR(optimizer=optimizer, step_size=step_size, gamma=gamma)
 
     if checkpoint:
         tqdm.write(f'Loading checkpoint...')
-        model, optimizer = load_from_checkpoint(checkpoint, model, optimizer=optimizer)
+        model, optimizer = load_from_checkpoint(checkpoint, model, optimizer=optimizer, inference=inference)
 
     return model, criterion, optimizer, lr_scheduler
 
 
-def net(net_params, num_channels, inference=False):
+def net(model_name: str,
+        num_bands: int,
+        num_channels: int,
+        dontcare_val: int,
+        num_devices: int,
+        train_state_dict_path: str = None,
+        pretrained: bool = True,
+        dropout_prob: float = False,
+        loss_fn: str = None,
+        optimizer: str = None,
+        class_weights: Sequence = None,
+        net_params=None,
+        conc_point: str = None,
+        coordconv_params=None,
+        inference_state_dict: str = None):
     """Define the neural net"""
```

Comment:
A lot more arguments are passed in net() function rather than only "params" dictionary.